### PR TITLE
update status of registeredcluster

### PR DESCRIPTION
**Pull request by leena-jawale**

### CODE REVIEW

File: controllers/cluster-registration/cluster_registration_controller.go [hunk 1]
```diff
@@ -96,22 +97,75 @@ func (r *RegisteredClusterReconciler) Reconcile(ctx context.Context, req ctrl.Re
...
+			regCluster.Status.ClusterClaims = clusterClaims
+		}
+		if managedClusterList.Items[0].Status.Version != (clusterapiv1.ManagedClusterVersion{}) {
+			version := managedClusterList.Items[0].Status.Version
+			regCluster.Status.Version = version
+		}
+		if err := r.Client.Status().Patch(ctx, regCluster, patch); err != nil {
+			return err
+		}
+	}
```
[the remark this answers is not in the document]
Comment (leena-jawale) [reply controllers/cluster-registration/cluster_registration_controller.go:144]: Yes makes sense. 
Also one more thing, we are copying managedcluster status which have managedcluster word in many places. like 
```
conditions:
  - lastTransitionTime: "2022-03-31T19:12:56Z"
    message: Accepted by hub cluster admin
    reason: HubClusterAdminAccepted
    status: "True"
    type: HubAcceptedManagedCluster
  - lastTransitionTime: "2022-03-31T19:22:00Z"
    message: Managed cluster joined
    reason: ManagedClusterJoined
    status: "True"
    type: ManagedClusterJoined
  - lastTransitionTime: "2022-03-31T19:22:00Z"
    message: Managed cluster is available
    reason: ManagedClusterAvailable
    status: "True"
    type: ManagedClusterConditionAvailable
```

Do we want to change this too registeredcluster ?

**Review by robinbobbitt — APPROVED** [review]

**Pull request merged**